drift-transport-netty/src/test/java/com/facebook/drift/transport/netty/server/TestDriftNettyServerTransport.java

```diff
@@ -190,14 +190,19 @@ private static int testOutOfOrderNotSupported(
                 // send second request, which will be blocked in the server because this client does not support out of order responses
                 // the only way to test this is with a sleep because the request is blocked inside of the server IO stack
                 sendLogRequest(22, messages, protocol);
-                assertNull(results.poll(1, SECONDS), "Second request future");
+                SettableFuture<Object> secondResult = results.poll(1, SECONDS);
+
                 assertFalse(firstResult.isDone());
 
+                // After upgrading to Netty (netty-all) to version >= 4.1.38, the logic changed so
+                // that the Second request future changed from [null] to an object in a PENDING state.
```

Comment:
Isn't request in `PENDING` mean the processing has already started. ?

Reply:
@ajaygeorge I believe you're right since PENDING actually means started to process; I'm checking with the Netty project. https://github.com/netty/netty/issues/10433

In the meantime, I've bumped up Netty to the highest version that works and documented this.

Reply:
At a glance, I don't think there's a bug here. I think the [commit you referenced](https://github.com/netty/netty/pull/9252/files) simply changes Netty to enqueue a future, where it wouldn't before. That future does not get completed.

You should be able to simply change this test bit to
```
SettableFuture<Object> secondResult = results.poll(1 /* or something sufficiently long */, SECONDS);
assertNotNull(secondResult, "expected a future to be enqueued for the second request");
assertFalse(secondResult.isDone());
assertFalse(firstResult.isDone());
```
And don't fetch from `results` anymore (like you do on line 205), you already have the second request's future.

If these tests succeed and the others do too, can we please upgrade to the most recent version, `4.1.91.Final`?